fabric-mining-level-api-v1/src/main/java/net/fabricmc/fabric/mixin/mininglevel/ShearsItemMixin.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.mininglevel;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.ShearsItem;
+
+import net.fabricmc.fabric.api.mininglevel.v1.FabricMineableTags;
+
+/**
+ * Adds support for {@link FabricMineableTags#SHEARS_MINEABLE}.
+ */
+@Mixin(ShearsItem.class)
+abstract class ShearsItemMixin {
+	@Inject(method = "isSuitableFor", at = @At("HEAD"), cancellable = true)
+	private void onIsSuitableFor(BlockState state, CallbackInfoReturnable<Boolean> info) {
+		if (state.isIn(FabricMineableTags.SHEARS_MINEABLE)) {
+			info.setReturnValue(true);
+		}
+	}
+
+	@Inject(method = "getMiningSpeedMultiplier", at = @At("RETURN"), cancellable = true)
+	private void onGetMiningSpeedMultiplier(ItemStack stack, BlockState state, CallbackInfoReturnable<Float> info) {
+		if (info.getReturnValueF() == 1.0f) { // if not caught by vanilla checks
+			if (state.isIn(FabricMineableTags.SHEARS_MINEABLE)) { // mimics MiningToolItem.getMiningSpeedMultiplier
+				// In vanilla 1.17, shears have three special mining speed multiplier values:
+				//   - cobweb and leaves return 15.0
+				//   - wool returns 5.0
+				//   - vines and glow lichen return 2.0
+				// As the most "neutral" option out of these three,
+				// we'll use 5.0 as it's not extremely fast nor extremely slow.
+				info.setReturnValue(5.0f);
```

Comment:
Can we just split the 3 mining speeds into 3 tags? And we specify which tags have priority?

Reply:
Hmm, that's an interesting idea. I could imagine an additional two tags (`mineable/shears/weak` and `mineable/shears/strong`) that would be used alongside this, but I'd like another opinion before changing this - a single value is still simpler.

Reply:
There should be a way for blocks to implement this switch themselves imo.

Reply:
So how should the blocks implement it? I think there are three reasonable options:

1. Tags (my preferable option)
2. A callback (feels like bloat tbh)
3. An interface to implement on blocks (takes out the data-driven aspect of mining levels)

Reply:
Does it make sense for a block's mining speed multiplier to depend on the tool that is used? I feel like blocks should have control over this maybe?

Reply:
Perhaps that would be a `FabricBlockSettings`-added setting? Think `BiFunction<BlockState, ItemStack, Double> dynamicMiningSpeedMultiplier` or something like that.

Reply:
> Does it make sense for a block's mining speed multiplier to depend on the tool that is used? I feel like blocks should have control over this maybe?

The tool attribute API supports this from the tool side. I can add the block stuff too, but anything touching mining levels also has to go through tool attributes to support dynamic tools - basically adding everything twice.

Reply:
Imo mining levels and breaking speed are different concerns. A suitable mining level would just delegate to the block settings to grab the speed (in fact that's how tools work). Unfortunately mojang had to do it differently for swords and shears.

Reply:
Maybe we could go for a simple
```java
@Nullable
Double getShearsMiningSpeedMultiplier(BlockState state);
```
auto-invoking event? And the same for swords?

Reply:
It sucks that it's hard-coded, so maybe we should have this event have a tool tag parameter? Not sure how useful it'd be, I think having different breaking speeds for different kinds of tools would be useful. Auto-invoking events allow both easy implementation on the Block class, and more flexible implementations with regular event handlers.

Reply:
Mining levels and breaking speeds are very coupled in the current Fabric API structure since the (broken) object builders method of setting required mining levels is tied to the tool attribute module, which in turn is tied to breaking speeds.

I'm a bit wary of adding mining speed events to this PR since it feels out of scope. That event would also have to be *in* the mining level API (since that's what implements the shear mining levels in this PR), which disagrees with the idea that "mining levels and breaking speed are different concerns".